pkg/sink/kafka/sarama_factory.go

```diff
@@ -75,11 +76,25 @@ func (f *saramaFactory) AdminClient(ctx context.Context) (ClusterAdminClient, er
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return &saramaAdminClient{
+	a := saramaAdminClient{
 		client:     client,
 		admin:      admin,
 		changefeed: f.changefeedID,
-	}, nil
+		done:       make(chan struct{}),
+	}
+	idleMs, err := a.GetBrokerConfig(ctx, BrokerConnectionsMaxIdleMsConfigName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	idleMsInt, err := strconv.Atoi(idleMs)
+	if err != nil || idleMsInt <= 0 {
+		log.Warn("invalid broker config",
+			zap.String("configName", BrokerConnectionsMaxIdleMsConfigName), zap.String("configValue", idleMs))
+		return nil, errors.Trace(err)
+	}
+	keepConnAliveInterval = time.Duration(idleMsInt/2) * time.Millisecond
+	go keepConnAlive(client, a.done)
```

Comment:
Do not spawn goroutine here.

AdminClient is called by the ddl sink and dml sink.

it's used to initialize the topicManager, this method refresh metadata each 10 minutes, add one branch to the backgroundRefreshMeta method to send heartbeat if necessary.